sympy/tensor/tensor.py

```diff
@@ -2997,6 +3013,46 @@ def _eval_rewrite_as_Indexed(self, tens, indices):
         expr = Indexed(tens.args[0], *index_symbols)
         return self._check_add_Sum(expr, index_symbols)
 
+    def _eval_partial_derivative(self, s):  # type: (Tensor) -> Expr
+
+        if not isinstance(s, Tensor):
+            return S.Zero
+        else:
+
+            # @a_i/@a_k = delta_i^k
+            # @a_i/@a^k = g_ij delta^j_k
+            # @a^i/@a^k = delta^i_k
+            # @a^i/@a_k = g^ij delta_j^k
+            # TODO: if there is no metric present, the derivative should be zero?
+
+            if self.head != s.head:
+                return S.Zero
+
+            # if heads are the same, provide delta and/or metric products
+            # for every free index pair in the appropriate tensor
+            # assumed that the free indices are in proper order
+            # A contravariante index in the derivative becomes covariant
+            # after performing the derivative and vice versa
+
+            kronecker_delta_list = []
+            for iself, iother in zip(self.free_indices, s.free_indices):
+                if iself.tensor_index_type is not iother.tensor_index_type:
```

Comment:
`!=` should be used.

`is not` checks if the objects occupy the same position in the computer memory... that is not a mathematical equality.

Reply:
Oh sorry, I will change that.